sklearn/feature_selection/mutual_info_.py

```diff
@@ -0,0 +1,447 @@
+# Author: Nikolay Mayorov <n59_ru@hotmail.com>
+# License: 3-clause BSD
+from __future__ import division
+
+import numpy as np
+from scipy.sparse import issparse
+from scipy.special import digamma
+
+from ..externals.six import moves
+from ..metrics.cluster.supervised import mutual_info_score
+from ..neighbors import NearestNeighbors
+from ..preprocessing import scale
+from ..utils import check_random_state
+from ..utils.validation import check_X_y
+from ..utils.multiclass import check_classification_targets
+
+
+def _compute_mi_cc(x, y, n_neighbors):
+    """Compute mutual information between two continuous variables.
+
+    Parameters
+    ----------
+    x, y : ndarray
+        Samples of random variables, 1-d arrays of identical shape.
+
+    n_neighbors : int
+        Number of nearest neighbors to search for each point, see [1]_.
+
+    Returns
+    -------
+    mi : float
+        Estimated mutual information. If it turned out to be negative it is
+        replace by 0.
+
+    Notes
+    -----
+    True mutual information can't be negative. If its estimate by a numerical
+    method is negative, it means (providing the method is adequate) that the
+    mutual information is close to 0 and replacing it by 0 is a reasonable
+    strategy.
+
+    References
+    ----------
+    .. [1] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
+           information". Phys. Rev. E 69, 2004.
+    """
+    n_samples = x.size
+
+    x = x.reshape((-1, 1))
+    y = y.reshape((-1, 1))
+    xy = np.hstack((x, y))
+
+    nn = NearestNeighbors(metric='chebyshev', n_neighbors=n_neighbors)
+
+    nn.fit(xy)
```

Comment:
I use `kd_tree` because it allows passing an array for `radius`. For the first `fit` I rely on `NearestNeighbors` to choose the fastest algorithm.